src/window/window_wrapper/mod.rs

```diff
@@ -222,7 +222,13 @@ pub fn start_loop(
         .with_decorations(!SETTINGS.get::<CmdLineSettings>().frameless);
 
     #[cfg(target_os = "linux")]
-    let winit_window_builder = winit_window_builder.with_app_id("Neovide".to_string());
+    let winit_window_builder = winit_window_builder
+        .with_app_id(SETTINGS.get::<CmdLineSettings>().wayland_app_id.to_string());
+    #[cfg(target_os = "linux")]
+    let winit_window_builder = winit_window_builder.with_class(
+        SETTINGS.get::<CmdLineSettings>().x11_class.to_string(),
+        SETTINGS.get::<CmdLineSettings>().x11_class.to_string(),
```

Comment:
with_class takes two args, `class` and `instance`, i think we should customize only `class`, and leave other as default "neovide".

Also, i think you can avoid having to prefix both calls with `#[cfg(target_os = "linux")]`, if you put this calls inside block, like this

```
#[cfg(target_os = "linux")]
{
  let winit_...
  let winit_...
}
```

Reply:
this block does not work for me

Reply:
```
    #[cfg(target_os = "linux")]
    let winit_window_builder = winit_window_builder
        .with_app_id(SETTINGS.get::<CmdLineSettings>().wayland_app_id)
        .with_class(
            SETTINGS.get::<CmdLineSettings>().x11_wm_class,
            "Neovide".to_string(),
        );
```

Try this.

Reply:
seems like this makes it ignore `--x11_class` :thinking:

Reply:
![image](https://user-images.githubusercontent.com/301015/126071038-29e6aacf-b9b6-43b0-8d23-5bfe17093667.png)

are you sure? works for me

Reply:
compiling right now, i missed some arguments in `cmd_line.rs`